pkgs/build-support/build-bazel-package/default.nix

```diff
@@ -50,13 +53,59 @@ let
   fArgs = removeAttrs args [ "buildAttrs" "fetchAttrs" "removeRulesCC" ];
   fBuildAttrs = fArgs // buildAttrs;
   fFetchAttrs = fArgs // removeAttrs fetchAttrs [ "sha256" ];
+  bazelCmd = { cmd, additionalFlags, targets }:
+    if targets == [ ] then "" else
+      ''
+          # Bazel sandboxes the execution of the tools it invokes, so even though we are
+          # calling the correct nix wrappers, the values of the environment variables
+          # the wrappers are expecting will not be set. So instead of relying on the
+          # wrappers picking them up, pass them in explicitly via `--copt`, `--linkopt`
+          # and related flags.
+          #
+          copts=()
+          host_copts=()
+          linkopts=()
+          host_linkopts=()
+          if [ -z "''${dontAddBazelOpts:-}" ]; then
+            for flag in $NIX_CFLAGS_COMPILE; do
+              copts+=( "--copt=$flag" )
+              host_copts+=( "--host_copt=$flag" )
+            done
+            for flag in $NIX_CXXSTDLIB_COMPILE; do
+              copts+=( "--copt=$flag" )
+              host_copts+=( "--host_copt=$flag" )
+            done
+            for flag in $NIX_LDFLAGS; do
+              linkopts+=( "--linkopt=-Wl,$flag" )
+              host_linkopts+=( "--host_linkopt=-Wl,$flag" )
+            done
+          fi
+
+          BAZEL_USE_CPP_ONLY_TOOLCHAIN=1 \
+          USER=homeless-shelter \
```

Comment:
homeless-shelter is not really a user but just a none existing directory where HOME points to by default and it should stay none existing. Why do we need to set this? If we need a HOME this should be done like ``export HOME=$(mktemp -d)

Reply:
There was a comment explaining this trick but not in every places it was used (I did not write this code, only moved it around).

```
Bazel computes the default value of output_user_root before parsing the flag.
The computation of the default value involves getting the $USER from the environment.
```
Since this is used in two places (and factorizing complicates things a bit) I moved the comment to the end of the file and redirected to it.